File: convert-pdf-to-images.js
// PDF 轉圖片腳本
// 使用 pdf-poppler 將 PDF 轉換為圖片

const pdf = require('pdf-poppler');
const path = require('path');
const fs = require('fs');

const pdfPath = path.join(__dirname, '../Topo_Mountain_Racer_4_Deep_Dive.pdf');
const outputDir = path.join(__dirname, 'public/slides/topo-mtn-racer-4');

// 確保輸出目錄存在
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

const options = {
  format: 'png',
  out_dir: outputDir,
  out_prefix: 'slide',
  page: null, // 轉換所有頁面
  scale: 2.0, // 提高解析度
};

console.log('開始轉換 PDF 為圖片...');
console.log('PDF 路徑:', pdfPath);
console.log('輸出目錄:', outputDir);

pdf.convert(pdfPath, options)
  .then(() => {
    console.log('轉換完成！');
    // 重新命名檔案為有序的格式
    const files = fs.readdirSync(outputDir)
      .filter(f => f.startsWith('slide') && f.endsWith('.png'))
      .sort();
    
    files.forEach((file, index) => {
      const oldPath = path.join(outputDir, file);
      const newPath = path.join(outputDir, `topo-mtn-racer-4-${String(index + 1).padStart(2, '0')}.png`);
      fs.renameSync(oldPath, newPath);
      console.log(`已重新命名: ${file} -> ${path.basename(newPath)}`);
    });
    
    console.log(`\n成功轉換 ${files.length} 張圖片！`);
    console.log(`圖片位置: ${outputDir}`);
  })
  .catch((error) => {
    console.error('轉換失敗:', error);
    console.error('\n如果出現錯誤，請嘗試：');
    console.error('1. 安裝 poppler: brew install poppler');
    console.error('2. 或使用 Python: pip3 install pdf2image pillow');
  });

File: convert_pdf_to_images.py
#!/usr/bin/env python3
"""
將 PDF 轉換為圖片
需要安裝: pip3 install pdf2image pillow
"""

import os
import sys
from pathlib import Path

try:
    from pdf2image import convert_from_path
except ImportError:
    print("錯誤：需要安裝 pdf2image")
    print("請執行: pip3 install pdf2image pillow")
    sys.exit(1)

# 設定路徑
script_dir = Path(__file__).parent
pdf_path = script_dir.parent / "Topo_Mountain_Racer_4_Deep_Dive.pdf"
output_dir = script_dir / "public" / "slides" / "topo-mtn-racer-4"

# 確保輸出目錄存在
output_dir.mkdir(parents=True, exist_ok=True)

if not pdf_path.exists():
    print(f"錯誤：找不到 PDF 檔案: {pdf_path}")
    sys.exit(1)

print(f"PDF 路徑: {pdf_path}")
print(f"輸出目錄: {output_dir}")
print("開始轉換...")

try:
    # 轉換 PDF 為圖片 (300 DPI)
    images = convert_from_path(
        str(pdf_path),
        dpi=300,
        fmt='png'
    )
    
    print(f"成功讀取 {len(images)} 頁")
    
    # 儲存每頁為 PNG
    for i, image in enumerate(images, 1):
        filename = f"topo-mtn-racer-4-{i:02d}.png"
        filepath = output_dir / filename
        image.save(filepath, 'PNG', optimize=True)
        print(f"已儲存: {filename} ({image.size[0]}x{image.size[1]})")
    
    print(f"\n✅ 轉換完成！共 {len(images)} 張圖片")
    print(f"圖片位置: {output_dir}")
    
except Exception as e:
    print(f"錯誤: {e}")
    print("\n如果出現 poppler 相關錯誤，請執行:")
    print("  brew install poppler")
    sys.exit(1)

